vnext/Scripts/msbuildeval.ps1

```diff
@@ -0,0 +1,73 @@
+# Evaluate an MSBuild property name
+param(
+    [Parameter(Mandatory=$true)]
+    [String]$SolutionFile,
+    [Parameter(Mandatory=$true)]
+    [String]$ProjectFile,
+    [Parameter(Mandatory=$true)]
+    [String]$PropertyName
+)
+
+function Get-MSBuildPath {
+    $vsWhere = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe"
+    if (!(Test-Path $vsWhere)) {
+        throw "Unable to find vswhere.exe."
+    }
+    $vsPath = & $vsWhere -version 16.5 -property installationPath;
+    return "$vsPath\MSBuild\Current\Bin\";
+}
+
+function Get-MSBuildProperty {
+    param (
+        [Parameter(Mandatory=$true)]
+        [String]$SolutionPath,
+        [Parameter(Mandatory=$true)]
+        [String]$ProjectPath,
+        [Parameter(Mandatory=$true)]
+        [String]$PropertyName
+    )
+
+    # Identify location of MSBuild
+    $MsBuildPath = Get-MSBuildPath
+
+    # Load Microsoft.Build.dll into script
+    Add-Type -Path "$MsBuildPath\Microsoft.Build.dll" | Out-Null
+
+    # Build a temporary "metaproj" of the solution file so it can be processed
+    ${env:MSBUILDEMITSOLUTION} = 1
+    & $MsBuildPath\MSBuild.exe $SolutionPath | Out-Null
+
+    # Build a local project collection
+    $projectCollection =[Microsoft.Build.Evaluation.ProjectCollection]::new()
+
+    # Process solution
+    $solution = [Microsoft.Build.Evaluation.Project]::new("$SolutionPath.metaproj", $null, "Current", $projectCollection)
+
+    # Clean up "metaproj" files
+    Remove-Item "$SolutionPath.metaproj" | Out-Null
+    Remove-Item "$SolutionPath.metaproj.tmp" | Out-Null
+
+    # Evaluate all of the Solution* properties and save into a collection
+    $globalProps = New-Object 'System.Collections.Generic.Dictionary[String,String]'
+    $solution.Properties | ForEach-Object -Process {
+        if ($_.Name.StartsWith("Solution")) {
+            $globalProps.Add($_.Name, $_.EvaluatedValue)
+        }
+    }
+
+    # Process the project file (with the Solution* properties we calculated before)
+    $project = [Microsoft.Build.Evaluation.Project]::new("$ProjectPath", $globalProps, "Current", $projectCollection)
+
+    # Look for the specified PropertyName and evaluate it
+    $project.Properties | ForEach-Object -Process {
+        if ($_.Name -eq $PropertyName) {
+            Write-Host $_.EvaluatedValue
+            break
+        }
+    }
```

Comment:
Re multiple invocation, how many properties do we end up seeing? I.e. could we call once to dump all of them, then pick things out in the JS layer?

Reply:
I think Julio was hinting at this too.

Reply:
Yeah, I just wanted to start with something "self-contained" since the only immediate requirement is being able to enable autolinking to determine which version of WinUI to use (and potentially fix #8591).